sklearn/utils/estimator_checks.py

```diff
@@ -289,7 +330,29 @@ def _construct_instance(Estimator):
     return estimator
 
 
-def check_estimator(Estimator):
+def _generate_instance_checks(name, estimator):
+    """Generate instance checks."""
+    yield from ((estimator, partial(check, name))
+                for check in _yield_all_checks(name, estimator))
+
+
+def _generate_class_checks(Estimator, raise_on_error):
+    """Generate class checks."""
+    name = Estimator.__name__
+    yield (Estimator, partial(check_parameters_default_constructible, name))
+
+    try:
+        estimator = _construct_instance(Estimator)
```

Comment:
Previously this was run without a try except clause. Could you explain why we need it now? 

Also a lot of things could raise a type error, and if something does we are going to silently ignore it if `not raise_on_error`. Is it safe to do so?

Reply:
This PR was updated to rename parameter to `raise_on_skip` and only catch the `SkipTest` exception.

Here `raise_on_skip` is set to `not generate_only`. When `generate_only=False`, we have the same behavior as before, all exceptions will be raised. When `generate_only=True`, if the `_construct_instance` raises a `SkipTest`, no other checks will be generated.

Reply:
This is for the use case where:

```py
def test_generate_inside():
    for estimator, check in check_estimator(estimator, estimator_only=True):
        check(estimator)
```

Assuming `_construct_instance` raises `SkipTest`, we either allow this to raise `SkipTest` or not.

Reply:
It would raise `SkipTest` even even if `check_parameters_default_constructible` passed. But that is what we do now if we run `check_estimator(estimator, estimator=False)`, so maybe this is the desired behavior?

Reply:
Updated PR to remove the whole try except block.